modules/cdp-engine/src/lib.rs

```diff
@@ -1306,15 +1304,24 @@ impl<T: Config> RiskManager<T::AccountId, CurrencyId, Balance, Balance> for Pall
 	}
 
 	fn check_debit_cap(currency_id: CurrencyId, total_debit_balance: Balance) -> DispatchResult {
-		let hard_cap = Self::maximum_total_debit_value(currency_id);
+		let hard_cap = Self::maximum_total_debit_value(currency_id)?;
 		let total_debit_value = Self::get_debit_value(currency_id, total_debit_balance);
 
-		ensure!(total_debit_value <= hard_cap, Error::<T>::ExceedDebitValueHardCap,);
+		ensure!(total_debit_value <= hard_cap, Error::<T>::ExceedDebitValueHardCap);
 
 		Ok(())
 	}
 }
 
+// Returns a list of currently supported/configured collateral currency
+impl<T: Config> Get<Vec<CurrencyId>> for Pallet<T> {
```

Comment:
This is not super clear. Move this to a helper `fn collateral_currency_ids()` and use `iter_keys`, and make a new ` `struct CollateralCurrencyIds` to implement `Get<Vec<CurrencyId>>`